plugin/evm/message/syncable.go

```diff
@@ -14,78 +14,107 @@ import (
 	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
 )
 
-var _ block.StateSummary = &SyncSummary{}
+var _ Syncable = (*BlockSyncSummary)(nil)
 
-// SyncSummary provides the information necessary to sync a node starting
+type Syncable interface {
+	block.StateSummary
+	GetBlockNumber() uint64
+	GetBlockHash() common.Hash
+	GetBlockRoot() common.Hash
+}
+
+type SyncableParser interface {
+	ParseFromBytes(summaryBytes []byte, acceptImpl AcceptImplFn) (Syncable, error)
+}
+
+type AcceptImplFn func(Syncable) (block.StateSyncMode, error)
+
+// BlockSyncSummary provides the information necessary to sync a node starting
 // at the given block.
-type SyncSummary struct {
+type BlockSyncSummary struct {
```

Comment:
nit did we need to rename `SyncSummary` to `BlockSyncSummary`? 🤔 
It seems like unneeded changes, unless there is a reason?

Reply:
This is to distinguish between AtomicBlockSyncSummary